modin/pandas/test/dataframe/test_iter.py

```diff
@@ -257,6 +257,17 @@ def test___setattr__():
 
     df_equals(modin_df, pandas_df)
 
+    # Use case from issue #4577
+    pandas_df = pandas.DataFrame([[1]], columns=["col0"])
+    modin_df = pd.DataFrame([[1]], columns=["col0"])
+
+    pandas_df.col0 = [3]
+    modin_df.col0 = [3]
+
+    pandas_df.col0.ffill()
```

Comment:
We should add a separate test case where we mutate the replaced column in place with something simple like `iloc`, and then check that `df.col0` has the updated value. I think something like my example [here](https://github.com/modin-project/modin/issues/4577#issuecomment-1158015868) will work.

Reply:
Makes sense and that test seems to cover most things as well. I made changes to my PR to reflect this.